src/io/read_tree_consistentrees_ascii.c

```diff
@@ -377,18 +382,21 @@ int64_t load_forest_ctrees(const int32_t forestnr, struct halo_data **halos, str
     /* Fix flybys -> multiple roots at z=0 must be joined such that only one root remains */
     int status = fix_flybys(totnhalos, forest_halos, info, verbose);
     if(status != EXIT_SUCCESS) {
-        ABORT(status);
+        return -EXIT_FAILURE;
     }
 
 
     /* Entire tree is loaded in. Fix upid's (i.e., only keep 1-level halo hierarchy: FOF->subhalo */
     const int max_snapnum = fix_upid(totnhalos, forest_halos, info, verbose);
     if(max_snapnum < 0) {
-        ABORT(max_snapnum);
+        return -EXIT_FAILURE;
     }
     
     /* Now the entire tree is loaded in. Assign the mergertree indices */
-    assign_mergertree_indices(totnhalos, forest_halos, info, max_snapnum);
+    status = assign_mergertree_indices(totnhalos, forest_halos, info, max_snapnum);
+    if(status < 0) {
```

Comment:
Is this correctly `< 0`?  Please check and ensure it shouldn't be `status != EXIT_SUCCESS`.

Reply:
Yup - the `assign_mergertree*` function is actually returning an execution status but the callee *must* return a negative status. Fixed with this:
```c
    if(status != EXIT_SUCCESS) {
        /* Not successful but the exit status from this routine has to be negative */
        const int neg_status = status < 0 ? status:-status;
        return neg_status;
    }

```